src/pow.cpp

```diff
@@ -237,6 +239,18 @@ bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params
     return true;
 }
 
+/** Calculate compressed weak block difficulty from compressed strong block difficulty
+   FIXME: Figure out scheme to
+   My very personal preference: Median of per-strong-block votes of weak block difficulty
+   ratio. */
+uint32_t WeakBlockProofOfWork(uint32_t nBits) {
```

Comment:
Have there been other suggestions for calculating the weak block difficulty besides the one you mention here, i.e. median of strong block voted weak block difficulty?

Reply:
Yes. If you have good insights on this, I am all ears. Lately, I am thinking that this is, as a network parameter, something that could be set individually per node. What I will do now is the following:

Implement a static limit of 1/600th the strong block difficulty, below which weak blocks will be straight out rejected and banScore increased. The idea is that those kinds of blocks would be truly spam - it is unlikely that the network will find consensus on weakblocks within a second across the globe, especially given the exponential time distribution makes many blocks even faster. Which means, in my code, to just have a corresponding hard lower bound for weak blocks.

Then, I'll add a command line argument that allows to set a per-node threshold at which weak blocks will be forwarded and will default that to 20s for now. Below that, I'd consider but drop them.

This is nothing final, but Peter R. likes to incorporate weakblocks into gigablocks testing, and I like to use that opportunity as well and those are the values we thought would make sense during an informal discussion. From the results of that testing, I suspect we also might get better parameters for future deployments, including on the main chain.

Reply:
Sounds good. I think the static limit of 1/600th the strong block difficulty is a good floor for now: agree that reaching weakblock consensus at the 1-second interval is unlikely.